docs/ref/models/constraints.txt

```diff
@@ -68,3 +68,41 @@ ensures only one location can exist for each ``date``.
 .. attribute:: UniqueConstraint.name
 
 The name of the constraint.
+
+``condition``
+-------------
+
+.. attribute:: UniqueConstraint.condition
+
+A :class:`Q` object that specifies the condition you want the constraint to
+enforce.
+
+For example ``UniqueConstraint(fields=['email'], condition=Q(status='ACTIVE')``
+ensures your emails in ACTIVE status are unique.
+
+.. admonition:: Restrictions on PostgreSQL
+
+    PostgreSQL requires functions referenced in the condition to be be marked as
+    IMMUTABLE. Django doesn't validate this but PostgreSQL will error. This
+    means that functions such as :ref:`date-functions` and
+    :class:`~django.db.models.functions.Concat` aren't accepted. If you store
```

Comment:
I guess this note was copied from ref/models/indexes.txt but I'm not clear on why database functions can be used with Q objects (perhaps as lookups?).

Reply:
I can provide next abstract example: `models.UniqueConstraint(fields=['z1'], name='z_uniq', condition=models.Q(z1=Lower('z2')))`, eg. when you use it with two fields.

Reply:
Okay. Maybe it would be better to something like "The same database restrictions that apply to Index.condition apply to UniqueConstraint.condition." (rather than repeating the same 4 boxes)

Reply:
changed to `UniqueConstraint.condition has the same database restrictions that Index.condition has.`